### DISCUSSION TITLE
Signature Hints

### DISCUSSION THREAD

**Discussion by salkin-mada** [davidgranstrom/scnvim#241] [Q&A] [answered]
#238 

Yeah. This is something I have been not so happy with for a long time.
Before `scnvim` had a working _signature hint pop-up_ I actually did a PR with a signature hint float window which you could close or make it stick at will, in addition to it automatic copying all args to a register **"** of your choice (by config). I have been missing these features quite a lot.
I dont know why but the PR was never worked with or merged.
And then afterwards signatures hint was implemented in the version it is today.

**Comment by davidgranstrom** [comment, marked as answer]
The behaviour you are describing should be possible now since https://github.com/davidgranstrom/scnvim/pull/242 was merged (in conjunction with setting the `close_events` config option to an empty table, mentioned in the linked issue) . 

I'm curious about the automatic copying of arguments you mention, what is the use-case? That behaviour could perhaps be implemented as config option in the current version of the signature module?

Sorry about the PR not being picked up, I don't remember the details now since it was quite some time ago, possibly it was because I was being in the middle of the lua rewrite of the plugin.

**Reply by salkin-mada** [reply]
Ah yes, I remember it was the lua rewrite days. <3
I also think my PR had a tiny bit of drive-by-commit aura to it. Hehe.
The use case is nothing more than if we are inside `SomeClass.method(|)` and didn't trigger the snippet, but are too lazy to write this special methods 12 args by hand, then its easy to insert from said register. Not really really important. Just a detail which is doing a minimal upping of QOL.
